src/main/java/racingcar/domain/car/Cars.java

```diff
@@ -0,0 +1,68 @@
+package racingcar.domain.car;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import racingcar.util.RandomGenerator;
+import racingcar.view.input.error.InputError;
+import racingcar.view.input.error.InputIllegalArgumentException;
+
+public class Cars {
+
+    private static final String CARS_ARE_EMPTY = "차가 존재하지 않습니다.";
+    public static final String DISTANT_MARK = "-";
+
+    private final List<Car> cars = new ArrayList<>();
+
+    public Cars(List<String> carNames) {
+        validateCanRace(carNames);
+
+        for (String name : carNames) {
+            cars.add(new Car(name));
+        }
+    }
+
+    private void validateCanRace(List<String> names) {
+        if (names.size() < 2) {
+            throw new InputIllegalArgumentException(InputError.NOT_EXCEED_MINIMUM_AMOUNT);
+        }
+    }
+
+    public List<String> findWinners() {
+        int maxDistance = findMaxDistance();
+
+        return cars.stream()
+                .filter(car -> car.getDistance() == maxDistance)
+                .map(Car::getName)
+                .collect(Collectors.toList());
+    }
+
+    private int findMaxDistance() {
+
+        return cars.stream()
+                .mapToInt(Car::getDistance)
+                .max()
+                .orElseThrow(() -> new IllegalArgumentException(CARS_ARE_EMPTY));
+    }
```

Comment:
저도 이번 미션을 통해 학습하게 되었는데요! Car객체의 내부 값을 꺼내서 Cars에서 처리를 해버리면 Car객체는 수동적인 객체가 된다고 해요! 이렇게 되면 Car내부 구현이 변경되면 Cars객체에까지 영향이 전파가 되기도 하구요! Car객체 스스로가 판단할 수 있도록 isGreaterThan(int maxDistance)라는 느낌의 메서드를 Car객체에 구현하고 Cars에서 메세지를 보내보는 방법은 어떠실까요?? 😄

Reply:
전체적으로 이번 리뷰를 보면서 느낀거였는데,
제 Cars가 유독 뚠뚠한 이유가 이와 같았던 것 같네요😭

최대한 자료구조를 많이 사용해보자는 생각 때문에 수동적으로 만들어버렸던 것 같단 생각이 들어요.
`isGreaterThan(int maxDistance)` 를 사용한다면 최댓값을 변수로 두어 비교하는 로직으로 구현을 말씀하시는 걸까요?!
방법이 궁금합니다! 🥰

Reply:
스트림을 잘 사용하지 못해서 반복문으로 예시를 들면,

```suggestion
    private int findMaxDistance() {
          int maxDistance = 0;
          for(Car car : cars) {
              maxDistance = car.getGreaterDistance(maxDistance);
          }
          return maxDistance;
      }
        
       // Car
       int getGreaterDistance(int maxDistance) {
        if (distance > maxDistance) {
            return distance;
        }
        return maxDistance;
    }
```

저는 이런식으로 할 수 있다고 생각했는데, 막상 구현을 해보니 조금은 어색한 느낌이 들기도 하네요..! 객체 지향은 어려운 것 같아요 😂

Reply:
훨씬 보기 편하다는 생각이 드네요
이후 해당 `maxDistance` 값을 가진 것을 List에서 `name`을 찾아 
우승자를 공표하는 것일까요?!
좋은 생각 같습니다!! 감사합니다! 3주차때는 조금 더 객체가 스스로 움직일 수 있도록 생각해봐야겠네요!! 🥰

Reply:
저도 2주차 미션때 cars list로 관리하려다 보니 로직이 복잡해지고, getter() 가 많이 사용되어서 고민이었습니다! 저도 다음주차땐 객체가 스스로 움직일 수 있도록 구현해봐야겠습니다!!